compiler/ballerina-lang/src/main/java/io/ballerina/projects/JarResolver.java

```diff
@@ -155,7 +166,16 @@ private void addPlatformLibraryPaths(PackageContext packageContext,
 
                 if (existingVersion.compareTo(newVersion) >= 0) {
                     if (!newEntry.packageName().orElseThrow().startsWith(ProjectConstants.BALLERINA_ORG)) {
-                        // TODO: issue a warning. For this we need to design diagnostic reporting in JarResolver
+                        // Issue warning for conflicting jars
+                        JBallerinaBackend.JarConflict conflict = new JBallerinaBackend.JarConflict(existingEntry,
+                                newEntry, Collections.emptyList());
+                        String conflictWarning = "Detected conflicting jar files: "
```

Comment:
Shall we rephrase this as `detected versions v1, v2 for <groupId>/<artifactId>`. Contnuing with v2. Also we don't need to consider this as a conflict and so listing the conflicted classes are not necessary.

Reply:
Added